File: iota_basic_transaction.py
# Import all functions for the iota module
# Install via pip with C bindings for faster crypto functions: 'pip install pyota[ccurl]'
from iota import *
import random

# A seed is used as your 'private key' so never share it!
SEED = b"XLPQORJUMCYE9PAYXTNNMATSVAAUTFMNJPSBYWU9MAQWAYICBTVADUESPOOTY9OBSNLZWXUOFVRMYYBGN"

# Address to send the transaction to
RECIPIENT_ADDRESS = b"IOTAHACKATHONIOTAHACKATHONIOTAHACKATHONIOTAHACKATHONIOTAHACKATHONIOTAHACKATHONIOTXBGOLUNQD"

# Possibles nodes
NODES = ['http://node02.iotatoken.nl:14265', 'http://node04.iotatoken.nl:14265', 'http://node06.iotatoken.nl:14265']

# Random index for node selection
RANDOM = random.randint(0, len(NODES)-1)

# Create IOTA instance directly with Node (provider)
# This is node is your entry point to the Tangle
api =\
  Iota(
    NODES[RANDOM],
    seed = SEED
  )

# For more information, see :py:meth:`Iota.send_transfer`.
result = api.send_transfer(
  depth = 100,

  # One or more :py:class:`ProposedTransaction` objects to add to the
  # bundle.
  transfers = [
    ProposedTransaction(
      # Recipient of the transfer.
      address =
        Address(
          RECIPIENT_ADDRESS,
        ),

      # Amount of IOTA to transfer.
      # This value may be zero.
      value = 0,

      # Optional tag to attach to the transfer.
      tag = Tag(b'IOTAHACKATHONGRONINGEN99999'),

      # Optional message to include with the transfer.
      message = TryteString.from_string('My first transaction with IOTA!'),
    ),
  ],
)

print(result['bundle'].hash)

File: iota_basic_transaction.js
// Import the IOTA library.
// Install with this command: 'npm install iota.lib.js'
var IOTA = require('iota.lib.js');

// List of possible nodes
var nodes = ['http://node02.iotatoken.nl:14265', 'http://node04.iotatoken.nl:14265', 'http://node06.iotatoken.nl:14265'];

// Random index for node selection
var random = Math.floor(Math.random() * nodes.length);

// Create IOTA instance directly with Node (provider)
// This is node is your entry point to the Tangle
var iota = new IOTA({
    'provider': nodes[random]
});

// A seed is used as your 'private key' so never share it!
const seed = '9YTPIWYMNCQ9HQVCOPUSEUEJQNHDPMFZGSJDIIILSBTDHFTPXIOUOGAPAEIE9TIGHRONRPDGLJOPMSGFX'; // KEEP IT SECURE!!

// Create a message to attach to your transaction
// and convert it to Trytes (data format used by IOTA)
let message = 'Test 1';
let messageTrytes = iota.utils.toTrytes(message);

// The receiving address (Harm's wallet!)
let address = 'STTENJJESAEJBMTNFJSFN9TUSQDLHTKGPFKLFSALUCU9YOUKTFUXJBDBDILOHTPCRMRVCLXYLFGOCXFPXYXGSRJTCC';

// Construct the transfer bundle with address, 
// value (amount of IOTA), a tag and a message
var transfer = [{
    'address': address,
    'value': 10,
	'tag': 'IOTAHACKATHONGRONINGEN',
	'message': messageTrytes,
}];

// Send transfer to the Tangle. Depending on the Node config of the connected Node
// you also do the PoW at this point. The magic numbers can stay there for the Mainnet
iota.api.sendTransfer(seed, 9, 14, transfer, function(e, bundle) {
    if (e) throw e;
    console.log("Successfully sent your transfer: ", bundle);
});



